### PR TITLE
Add accessibility section

### LINKED ISSUE #70: Requesting Accessibility Considerations section
An accessibility review was requested of the APA as part of our role in performing horizontal review of W3C documents for accessibility concerns. The Device Posture APA was reviewed by APA member Paul Grenier who proposed the following comments that were approved by the group.

1. Per previous discussions, the group would prefer to see summary accessibility considerations organized under a heading at the same level as Security and Privacy Considerations.
2. In the Accessibility Considerations section, the group would like to see an extrapolation of 1.3.4 Orientation <https://www.w3.org/WAI/WCAG21/Understanding/orientation.html> applied to device posture until it can be addressed in those documents. For example, "Content does not restrict its view and operation to a single display orientation <https://www.w3.org/TR/screen-orientation/> or single device posture <https://www.w3.org/TR/device-posture/#posture-modes>, unless a specific display arrangement is essential <https://www.w3.org/WAI/WCAG21/Understanding/orientation.html#dfn-essential>

### PR DESCRIPTION
**Pull request by darktears**
Closes #70

***
<a href="https://pr-preview.s3.amazonaws.com/darktears/device-posture/pull/127.html" title="Last updated on Mar 12, 2024, 2:08 PM UTC (02a2ea3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/device-posture/127/5ac8c25...darktears:02a2ea3.html" title="Last updated on Mar 12, 2024, 2:08 PM UTC (02a2ea3)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -606,6 +606,54 @@ <h4>
         </section>
       </section>
     </section>
+    <section>
+      <h2>
+        Accessibility considerations
+      </h2>
+      The Device Posture API main focus is to improve the user experience. There are three ways
```
Comment (kenchris) [inline index.html]: The main focus of...

File: index.html [hunk 2]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+      in which applications that build on the API can positively impact accessibility.
+      <ol>
+        <li>Designing and making applications which does not place content in the fold/hinge
```
Comment (kenchris) [inline index.html]: do (plural)

File: index.html [hunk 3]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          area, especially buttons. This area is typically hard to interact with with fingers.
```
Comment (kenchris) [inline index.html]: with, with? Maybe reword. Why is it hard? because touch work worse, or because its not stiff?

File: index.html [hunk 4]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+        </li>
+        <li>Designing and making applications which does not put a large contiguous content
```
Comment (kenchris) [inline index.html]: do*

File: index.html [hunk 5]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          (like a video or a picture) spanning across the fold/hinge area. This area slightly
+          distorts content and colors.
```
Comment (kenchris) [inline index.html]: ..in the case it is semi-folded

File: index.html [hunk 6]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+        </li>
+        <li>Designing and making applications which utilize the screen estate better by
+          providing a split UI, allowing the application to provide a differentiated
```
Comment (kenchris) [inline index.html]: split UI - a user interface there the content is split up across the screen segments...

File: index.html [hunk 7]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+      in which applications that build on the API can positively impact accessibility.
+      <ol>
+        <li>Designing and making applications which does not place content in the fold/hinge
+          area, especially buttons. This area is typically hard to interact with with fingers.
+        </li>
+        <li>Designing and making applications which does not put a large contiguous content
+          (like a video or a picture) spanning across the fold/hinge area. This area slightly
+          distorts content and colors.
+        </li>
+        <li>Designing and making applications which utilize the screen estate better by
```
Comment (kenchris) [inline index.html]: utilize - use

File: index.html [hunk 8]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          providing a split UI, allowing the application to provide a differentiated
+          and better interface.
+        </li>
+      </ol>
+        When using this API it is important to consider these opportunities with accessibility
```
Comment (kenchris) [inline index.html]: these? the above? the below?

File: index.html [hunk 9]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+        in mind. Here are few concrete examples:
+      <ul>
+        <li><strong>UI elements placement:</strong> A typical placement logic for dialogs
```
Comment (kenchris) [inline index.html]: Placement of UI elements

File: index.html [hunk 10]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          is to center them horizontally and vertically but in a device folded
+          with a symmetrical fold they may end up in the middle of the fold area
+          making buttons hard to interact.
```
Comment (kenchris) [inline index.html]: making it harder to interact with, for instance clicking on buttons. It might also make it harder to read text due to distorsion and light reflection

File: index.html [hunk 11]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+        </li>
+        <li><strong>UI elements in fullscreen:</strong> When resizing an element in
```
Comment (kenchris) [inline index.html]: fullscreen mode

File: index.html [hunk 12]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          fullscreen, a typical logic would expand that element to the width and
+          the height of the {{Window}}. If the device is folded, the element will be
+          laid out across the fold making a subpar user experience.
+          An alternative will be to choose to display that element either on top or
```
Comment (kenchris) [inline index.html]: will??

File: index.html [hunk 13]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          below the fold area.
+        </li>
+        <li><strong>Improving the user experience:</strong> Using this API when the
+          device is folded allows the possibility to create differentiated user experiences.
```
Comment (kenchris) [inline index.html]: With this API it is possible to create a different and adapted user experience when the device is folder, allowing for a better user experience and improved accessibility

File: index.html [hunk 14]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          For example, when watching a video and the device is folded, the video could
+          be displayed on top of the fold while comments could be shown below the fold.
```
Comment (kenchris) [inline index.html]: while the comment section .... below the fold by default

File: index.html [hunk 15]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          Another example would be a video conferencing application showing the video
+          feed on top of the fold and the chat below the fold.
+        </li>
+        <li><strong>Content shouldn't be designed for one {{posture}} only:</strong>
```
Comment (kenchris) [inline index.html]: for just one posture

File: index.html [hunk 16]
```diff
@@ -606,6 +606,54 @@ <h4>
...
+          The idea of foldable devices is their versatility and the ability for the user
+          to change the posture as they see fit. Enforcing a certain UI paradigm may not
+          fit the user need or particular setup of his device.
+        </li>
```
Comment (kenchris) [inline index.html:661]: With accessibility, it is important to not always choose for the user as they might have other needs. So this allows for better defaults but we should maybe mention that making things configurable is always great

File: index.html [hunk 17]
```diff
@@ -606,6 +606,61 @@ <h4>
...
+    <section>
+      <h2>
+        Accessibility considerations
+      </h2>
+      The main focus of the Device Posture API is to improve the user experience. There are three ways
+      in which applications that build on the API can positively impact accessibility.
+      <ol>
+        <li>Designing and making applications which do not place content in the fold/hinge
+          area, especially buttons. This area is typically hard to interact with with fingers
+          because the curvature of the fold make touch less precise or impossible.
```
Comment (rakuco) [inline index.html:618]:
```suggestion
          area, especially buttons. This area is typically hard to interact with using fingers
          because the curvature of the fold makes touch less precise or impossible.
```

File: index.html [hunk 18]
```diff
@@ -606,6 +606,61 @@ <h4>
...
+        </li>
+        <li>Designing and making applications which do not put a large contiguous content
```
Comment (rakuco) [inline index.html:620]:
```suggestion
        <li>Designing and making applications which do not put large, contiguous content
```

File: index.html [hunk 19]
```diff
@@ -606,6 +606,61 @@ <h4>
...
+          (like a video or a picture) spanning across the fold/hinge area if the device is folded.
```
Comment (rakuco) [inline index.html:621]:
```suggestion
          (such as a video or a picture) spanning across the fold/hinge area if the device is folded.
```

File: index.html [hunk 20]
```diff
@@ -606,6 +606,61 @@ <h4>
...
+          This area slightly distorts content and colors.
+        </li>
+        <li>Designing and making applications which use the screen estate better by
+          providing a split UI - a user interface where the content is split up across
+          the screen segments -, allowing the application to provide a differentiated
```
Comment (rakuco) [inline index.html:626]:
```suggestion
          providing a split UI (a user interface where the content is split up across
          the screen segments), allowing the application to provide a differentiated
```

File: index.html [hunk 21]
```diff
@@ -606,6 +606,61 @@ <h4>
...
+          and more powerful interface.
+        </li>
+      </ol>
+        When using this API it is important to consider the opportunities above with accessibility
+        in mind. Here are few concrete examples:
+      <ul>
+        <li><strong>Placement of UI elements:</strong> A typical placement logic for dialogs
+          is to center them horizontally and vertically but in a device folded
```
Comment (rakuco) [inline index.html:634]:
```suggestion
          is to center them horizontally and vertically, but in a device folded
```

File: index.html [hunk 22]
```diff
@@ -606,6 +606,61 @@ <h4>
...
+          with a symmetrical fold they may end up in the middle of the fold area
+          making it harder to interact with, for instance clicking on buttons.
+          It might also make it harder to read text due to distortion and light
+          reflection.
+        </li>
+        <li><strong>Fullscreen mode:</strong> When resizing an element in
+          fullscreen, a typical logic would expand that element to the width and
+          the height of the {{Window}}. If the device is folded, the element will be
+          laid out across the fold making a subpar user experience.
```
Comment (rakuco) [inline index.html:643]:
```suggestion
          laid out across the fold, leading to a subpar user experience.
```

**Review by kenchris — APPROVED** [review]

**Review by rakuco — COMMENTED** [review]
I've added some inline comments with proof-reading suggestions.

Other than that: I see no mention of point (2) of #70, was it added elsewhere?

I'd also thank the APA (or the specific reviewer) in the Acknowledgments section.
